From 66edfa88a1f72d120c5b99d54aada1b6810239b0 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 25 Mar 2011 09:36:59 -0400
Subject: [PATCH] NetworkStatus: Adapt to nm_client_activate_connection api change

It now takes a connection directly instead of the object path associated
with a connection.
---
 js/ui/status/network.js |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/js/ui/status/network.js b/js/ui/status/network.js
index e57349a..7990332 100644
--- a/js/ui/status/network.js
+++ b/js/ui/status/network.js
@@ -379,7 +379,7 @@ NMDevice.prototype = {
         // pick the most recently used connection and connect to that
         // or if no connections ever set, create an automatic one
         if (this._connections.length > 0) {
-            this._client.activate_connection(this._connections[0].connection.path, this.device, null, null);
+            this._client.activate_connection(this._connections[0].connection, this.device, null, null);
         } else if (this._autoConnectionName) {
             let connection = this._createAutomaticConnection();
             this._client.add_and_activate_connection(connection, this.device, null, null);
@@ -552,10 +552,9 @@ NMDevice.prototype = {
     },
 
     _createConnectionItem: function(obj) {
-        let path = obj.connection.path;
         let item = new PopupMenu.PopupMenuItem(obj.name);
         item.connect('activate', Lang.bind(this, function() {
-            this._client.activate_connection(path, this.device, null, null);
+            this._client.activate_connection(obj.connection, this.device, null, null);
         }));
         return item;
     },
@@ -1001,7 +1000,7 @@ NMDeviceWireless.prototype = {
             for (let i = 0; i < bestApObj.accessPoints.length; i++) {
                 let ap = bestApObj.accessPoints[i];
                 if (this._connectionValidForAP(best, ap)) {
-                    this._client.activate_connection(best.path, this.device, ap.dbus_path, null);
+                    this._client.activate_connection(best, this.device, ap.dbus_path, null);
                     break;
                 }
             }
@@ -1136,7 +1135,7 @@ NMDeviceWireless.prototype = {
             let accessPoints = sortAccessPoints(accessPointObj.accessPoints);
             for (let i = 0; i < accessPoints.length; i++) {
                 if (this._connectionValidForAP(connection, accessPoints[i])) {
-                    this._client.activate_connection(connection.path, this.device, accessPoints[i].dbus_path, null);
+                    this._client.activate_connection(connection, this.device, accessPoints[i].dbus_path, null);
                     break;
                 }
             }
-- 
1.7.4.1

